synapse/handlers/federation.py

```diff
@@ -1300,8 +1300,37 @@ def on_invite_request(self, origin, pdu):
         context = yield self.state_handler.compute_event_context(event)
         yield self.persist_events_and_notify([(event, context)])
 
+        sender = UserID.from_string(event.sender)
+        target = UserID.from_string(event.state_key)
+        if (sender.localpart == target.localpart):
+            logcontext.run_in_background(
```

Comment:
for future reference, `synapse.metrics.background_process_metrics.run_as_background_process` is probably preferable to `run_in_background` for this sort of thing, as it will give the background thread its own metrics, and a fake "request-id" for the logs.